### DISCUSSION TITLE
PowerShell variable is empty when called ...

### DISCUSSION THREAD

**Discussion by ryanpcmcquen** [community/community#26171] [Actions]
<p>I am not sure what is going on here, but it seems that my PowerShell variables are empty, despite existing when I echo them, here is what I am running on the Windows container with the default shell:</p>
<pre><code>$VS_EDITION = dir 'C:\Program Files (x86)\Microsoft Visual Studio\' | ? -FilterScript {$_.Name -match "[0-9]*"} | Sort CreationTime | Select Name -First 1; $VCVARSALL_BIN = Get-ChildItem -Path "C:\Program Files (x86)\Microsoft Visual Studio\$($VS_EDITION.Name)\" -Filter 'vcvarsall.bat' -Recurse | Select FullName -First 1; $Env:CC = "`"C:\Program Files\LLVM\bin\clang.exe`""; cmd /k "`"$($VCVARSALL_BIN.FullName)`" x64 &amp;&amp; nmake"
</code></pre>
<p>For reference, here is the full workflow:</p>
<p><a href="https://github.com/ryanpcmcquen/basque/blob/master/.github/workflows/c.yml" rel="nofollow noopener">https://github.com/ryanpcmcquen/basque/blob/master/.github/workflows/c.yml</a></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>I tested your Powershell scripts in my side, I found when run it in the windows-latest runner, </p>
<pre><code>$VS_EDITION = dir 'C:\Program Files (x86)\Microsoft Visual Studio\' | ? -FilterScript {$_.Name -match "[0-9]*"} | Sort CreationTime | Select Name -First 1;
</code></pre>
<p>VS_EDITION will get the value :   @{Name=Installer}</p>
<p>Please see my scripts and logs:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/245b77edb54ce488445de9777f090ba92c6afd67.png" data-download-href="/uploads/short-url/5bD9S5gjNePTiyN2we7se3QZCXd.png?dl=1" title="vs edition.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095917-de5369c3-2d68-4fb0-9a39-57c01eafb26f.png" alt="vs edition.png" data-base62-sha1="5bD9S5gjNePTiyN2we7se3QZCXd" width="690" height="276" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/245b77edb54ce488445de9777f090ba92c6afd67_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">vs edition.png</span><span class="informations">1276×511 22.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The issue is that  <strong>-FilterScript {<span class="math">\_.Name -&lt;font color="#800000"&gt;match &lt;/font&gt;"[0-9]\*"}** is not right, it could not exclude Installer and Shared folder.&amp;nbsp; To use wildcard characters, please change -match to -like :&amp;nbsp;&amp;nbsp;**-FilterScript {</span>_.Name -like “[0-9]*”} </strong></p>
<p>I refered to this powershell document: <a href="https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/where-object?view=powershell-6#parameters" rel="nofollow noopener">https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/where-object?view=powershell-6#parameters </a></p>

**Comment by ryanpcmcquen** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a>, thank you so much! That did it. I had printed that value earlier and thought it was showing the year, but obviously something changed. I can’t tell you how many iterations of this I tried and I am <em>so</em> happy to have it fixed.</p>
